### PR TITLE
Fix test_draw_boxes

### PR DESCRIPTION
**Pull request by NicolasHug**
Looks like something changed (don't know what TBH) and this test is failing on new PRs like https://github.com/pytorch/vision/pull/3628 and https://github.com/pytorch/vision/pull/3624

This PR updates the reference image (EDIT: we also avoid the check if PIL_version < 8.2). The new image seems to be more correct as in the current one, the top pixel of the "b" letter seems to be separate from the rest https://github.com/pytorch/vision/blob/master/test/assets/fakedata/draw_boxes_util.png

EDIT: Closes https://github.com/pytorch/vision/issues/3637

### CODE REVIEW

**Comment by oke-aditya** [comment]
One possible reason can be that a [new version of PIL](https://github.com/python-pillow/Pillow/releases/tag/8.2.0) was released 3 days ago. Tests seem to start failing then.
Possible [PR](https://github.com/python-pillow/Pillow/pull/5183) that might have affected.

This might bring to one question.
Dependencies on any specific PIL version for plotting utils functionalities.

For `draw.Rectangle()`

As we can see there is extra width param in [stable Pillow](https://pillow.readthedocs.io/en/stable/reference/ImageDraw.html#PIL.ImageDraw.ImageDraw.rectangle)

While 4.1.1 does not have. Have a look [here](https://pillow.readthedocs.io/en/4.1.x/reference/ImageDraw.html#PIL.ImageDraw.PIL.ImageDraw.Draw.rectangle)

I see that in the `setup.py` file we have mentioned the minimum version [here](https://github.com/pytorch/vision/blob/master/setup.py#L70)

I verified once but I don't think so we have any other plotting dependency. But the results may vary based on PIL version.

**Comment by NicolasHug** [comment]
>  One possible reason can be that a new version of PIL was released 3 days ago

You're right, thanks. The failure is cause by the release of PIL 8.2 (tests still pass on master with 8.1).

Regarding the PIL dependency: it might indeed be time to bump the minimal version. But more importantly: it seems that we're currently only testing against the latest PIL version, which is a bit risky. Only testing the latest version means that if we're using a PIL API that was only introduced recently, our PIL dependency `pillow_ver = ' >= 4.1.1'` would be effectively broken but the CI wouldn't catch that. It'd be preferable to have at least 2 CI instances: one with the latest PIL and one with the minimal supported version.

@fmassa WDYT?

File: test/test_utils.py [hunk 1]
```diff
@@ -120,8 +123,11 @@ def test_draw_boxes(self):
             res = Image.fromarray(result.permute(1, 2, 0).contiguous().numpy())
             res.save(path)
 
-        expected = torch.as_tensor(np.array(Image.open(path))).permute(2, 0, 1)
-        self.assertTrue(torch.equal(result, expected))
+        if PILLOW_VERSION >= (8, 2):
```
Comment (NicolasHug) [inline test/test_utils.py:126]: Another solution is to set the condition to `if PILLOW_VERSION < (8, 2)` and not update the image.

**Review by fmassa — APPROVED** [review]
LGTM, thanks!

I'm ok adding one extra CI for an older version of PIL as well in a follow-up PR

**Pull request merged**